feat(GovDAO): add activity page to highlight inactive GovDAO's members

davd-gzl opened the pull request:
Fix: #4715 

Add a new sub-page "activity" on `/r/gov/dao/v3/impl` to check on Gnoweb the activities of voters.
It also add a timestamp when member join, and creation / closing timestamp of proposal to track correct members activities.

# Example of the activity page
```
# GovDAO Activity

## Members Activity
**[@1d5c47h6lta047h6lta047h6lta047h6lvhfyf2](/u/1d5c47h6lta047h6lta047h6lta047h6lvhfyf2)**: 1/1 (100.0%)
**[@1d5cjuv2lta047h6lta047h6lta047h6lvpa4mr](/u/1d5cjuv2lta047h6lta047h6lta047h6lvpa4mr)**: 0/1 (0.0%)
**[@1d5cjuvfwx9047h6lta047h6lta047h6l0640cd](/u/1d5cjuvfwx9047h6lta047h6lta047h6l0640cd)**: 0/1 (0.0%)
**[@1d5cjuvfwxyhrzh6lta047h6lta047h6lhwcnlj](/u/1d5cjuvfwxyhrzh6lta047h6lta047h6lhwcnlj)**: 0/1 (0.0%)
**[@1d5e47h6lta047h6lta047h6lta047h6l7arjx2](/u/1d5e47h6lta047h6lta047h6lta047h6l7arjx2)**: 0/1 (0.0%)
**[@1d5e97h6lta047h6lta047h6lta047h6l7w8k8v](/u/1d5e97h6lta047h6lta047h6lta047h6l7w8k8v)**: 0/1 (0.0%)
**[@1d5647h6lta047h6lta047h6lta047h6lpraph2](/u/1d5647h6lta047h6lta047h6lta047h6lpraph2)**: 0/1 (0.0%)
**[@1d5697h6lta047h6lta047h6lta047h6lpse9kv](/u/1d5697h6lta047h6lta047h6lta047h6lpse9kv)**: 0/1 (0.0%)
**[@1d5m97h6lta047h6lta047h6lta047h6ln6nnev](/u/1d5m97h6lta047h6lta047h6lta047h6ln6nnev)**: 0/1 (0.0%)
```

davd-gzl committed:
feat(govdao): timestamps on member and proposal status

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🟢 Pending initial approval by a review team member, or review from tech-staff


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: davd-gzl/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>Pending initial approval by a review team member, or review from tech-staff</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 Not (🔴 Pull request author is a member of the team: tech-staff)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 If
    ├── 🟢 Condition
    │   └── 🟢 Or
    │       ├── 🟢 User jefft0 already reviewed PR 4731 with state APPROVED
    │       ├── 🟢 At least 1 user(s) of the team tech-staff reviewed pull request
    │       └── 🟢 This pull request is a draft
    └── 🟢 Then
        └── 🟢 Not (🔴 This label is applied to pull request: review/triage-pending)

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

davd-gzl committed:
feat(govdao): add activity page and member activity tracking

davd-gzl committed:
fix: std split

davd-gzl committed:
feat(gov/dao): add working member activity page

davd-gzl committed:
fix: test

davd-gzl committed:
fix: test + remove useless comments

davd-gzl committed:
fix(types): remove useless check

jefft0 reviewed (approved):
Tests are included. CI checks pass. This adds reporting but doesn't affect the main GovDAO functionality. Ready for core devs to review.

moul reviewed (changes requested):
I like such data, but the implementation, using nested iterations without a pager, will become problematic.

davd-gzl committed:
fix(govdao): add configuration "Proposal to analyze"
feat(govdao): added tiers in activity

Davphla commented on examples/gno.land/r/gov/dao/v3/impl/types.gno:
```diff
@@ -82,7 +86,39 @@ func newProposalStatus(allowedToVote []string) *proposalStatus {
+		return activity, available_proposal
 	}
+	joinTime := member.JoinTime
+	gov.pss.Iterate("", "", func(id string, proposalValue interface{}) bool {
```
This isn't perfect, but there's 2 solutions for a lighter on-chain activity tracker:

- Sort by activities to highlight the inactive members -- This doesn't implement a pager since it wont optimize the sorting - https://github.com/gnolang/gno/pull/4731/commits/f98f8889aa51660ed5646047a7b479d94cb234e3
It adds a maximum number of proposals to analyze (Currently max: 20, this is an arbitrary value).
This implementation assume there won't be a huge amount of people in the GovDAO, which may be problematic in the long term.
<img width="1774" height="1204" alt="image" src="https://github.com/user-attachments/assets/c4cbe3c6-a664-4c42-9db0-125a4722eb13" />

- (Current implementation) Display non-sorted members and their activity with a pager, computed at the moment of actions. It is light and simple, but it does not highlight inactive members instantly. It requires an AVL tree for storing activities data - https://github.com/gnolang/gno/pull/4731/commits/3e69459cac9bb0cacd5f8a5a29d612612e533b9e
I think this is the most suitable implementation.
<img width="1690" height="1296" alt="image" src="https://github.com/user-attachments/assets/3e1e129b-b29a-4f59-8655-187b300b601d" />

davd-gzl committed:
fix(govdao): implement activity page using non-sorted stored data

davd-gzl committed:
refacto(govdao): remove dead code

davd-gzl committed:
refactor(govdao): remove unused time fields from proposalStatus

davd-gzl committed:
Merge remote-tracking branch 'origin/master' into 4715

# Conflicts:
#	examples/gno.land/r/gnoland/users/v1/filetests/z_0_prop1_filetest.gno
#	examples/gno.land/r/gnoland/users/v1/filetests/z_1_prop2_filetest.gno
#	examples/gno.land/r/gov/dao/v3/impl/render.gno
#	examples/gno.land/r/gov/dao/v3/loader/loader.gno

davd-gzl committed:
fix: update avl import path to v0 in activity.gno

Changed import from gno.land/p/nt/avl to gno.land/p/nt/avl/v0 to match
the package path rename on master.

davd-gzl committed:
Merge branch 'master' into 4715

davd-gzl commented:
Nice addition but not very important, it can be done off-chain
